guvnor-ala/guvnor-ala-ui/guvnor-ala-ui-client/src/main/java/org/guvnor/ala/ui/client/wizard/container/ContainerConfigPresenter.java

```diff
@@ -0,0 +1,275 @@
+import static org.guvnor.ala.ui.client.util.UIUtil.EMPTY_STRING;
+import static org.guvnor.ala.ui.client.util.UIUtil.trimOrGetEmpty;
+
+@Dependent
+public class ContainerConfigPresenter {
+
+    public interface View
+            extends UberElement<ContainerConfigPresenter> {
+
+        String getContainerName();
+
+        void setContainerName(final String containerName);
+
+        String getGroupId();
+
+        void setGroupId(final String groupId);
+
+        String getArtifactId();
+
+        void setArtifactId(final String artifactId);
+
+        String getVersion();
+
+        void setVersion(final String version);
+
+        void setArtifactSelectorPresenter(final IsElement artifactSelector);
+
+        void clear();
+
+        void setContainerNameStatus(final FormStatus status);
+
+        void setGroupIdStatus(final FormStatus status);
+
+        void setArtifactIdStatus(final FormStatus status);
+
+        void setVersionStatus(final FormStatus status);
+
+        void setContainerNameHelpText(final String containerNameHelpText);
+
+        void clearContainerNameHelpText();
+
+        void showFormError(final String error);
+
+        void clearFormError();
+    }
+
+    private final View view;
+
+    private final ArtifactSelectorPresenter artifactSelector;
+
+    private PopupHelper popupHelper;
+
+    private TranslationService translationService;
+
+    private final Caller<M2RepoService> m2RepoService;
+
+    private final ProvisioningClientValidationService validationService;
+
+    private List<String> alreadyInUseNames;
+
+    boolean containerNameValid = false;
+
+    @Inject
+    public ContainerConfigPresenter(final View view,
+                                    final ArtifactSelectorPresenter artifactSelector,
+                                    final TranslationService translationService,
+                                    final PopupHelper popupHelper,
+                                    final Caller<M2RepoService> m2RepoService,
+                                    final ProvisioningClientValidationService validationService) {
+        this.view = view;
+        this.artifactSelector = artifactSelector;
+        this.popupHelper = popupHelper;
+        this.translationService = translationService;
+        this.m2RepoService = m2RepoService;
+        this.validationService = validationService;
+    }
+
+    @PostConstruct
+    public void init() {
+        view.init(this);
+        artifactSelector.setArtifactSelectHandler(this::onArtifactSelected);
+        view.setArtifactSelectorPresenter(artifactSelector.getView());
+    }
+
+    public IsElement getView() {
+        return view;
+    }
+
+    public void clear() {
+        artifactSelector.clear();
+        view.clear();
+        containerNameValid = false;
+    }
+
+    public void setup(final List<String> alreadyInUseNames) {
+        this.alreadyInUseNames = alreadyInUseNames;
```

Comment:
i'd usually prefer for collection setter, create a new instance.. something like:

```java
this.alreadyInUseNames = new ArrayList<>(alreadyInUseNames);
```